packages/backend/src/queryBuilder.ts

```diff
@@ -807,34 +834,22 @@ export const buildQuery = ({
         ),
     ]);
 
-    const getJoinedTables = (tableNames: string[]): string[] => {
-        if (tableNames.length === 0) {
-            return [];
-        }
-        const allNewReferences = explore.joinedTables.reduce<string[]>(
-            (sum, joinedTable) => {
-                if (tableNames.includes(joinedTable.table)) {
-                    const newReferencesInJoin = parseAllReferences(
-                        joinedTable.sqlOn,
-                        joinedTable.table,
-                    ).reduce<string[]>(
-                        (acc, { refTable }) =>
-                            !tableNames.includes(refTable)
-                                ? [...acc, refTable]
-                                : acc,
-                        [],
-                    );
-                    return [...sum, ...newReferencesInJoin];
-                }
-                return sum;
-            },
-            [],
-        );
-        return [...allNewReferences, ...getJoinedTables(allNewReferences)];
-    };
+    const tableCompiledSqlWhere =
+        explore.tables[explore.baseTable].compiledSqlWhere;
```

Comment:
Since I'm using the new `compiledSqlWhere`, I realize this is a breaking change.

Before the change, we've been using `sqlWhere`, which is already compiled.

We need both the compiled and original versions to make the new filters work.

Something that I hadn't considered initially is that changes inside `exploreCompiler` happen at compile time, while `queryBuilder` changes occur at run time. So, whatever we end up compiling gets persisted in the DB.

Technically, the new solution adds the code to the right places, but since we only had the `sqlWhere` property, which is already a compiled version, it forces us to make a change that is not ideal to understand.

What we had before:
`sqlWhere` -> compiled SQL

Solution in the PR:
`sqlWhere` -> now contains uncompiled SQL (breaking change)
`compiledSqlWhere` -> contains compiled SQL

Proposed solution:
`sqlWhere` -> stays as compiled SQL
`uncompiledSqlWhere` -> will now hold uncompiled SQL

Let me know if you know any way to make this less confusing than it is.

Reply:
I agree with the proposed solution, seems to me like a good compromise to avoid a breaking change in the compiled tables 👌